engine/flink/components/base/src/it/scala/pl/touk/nussknacker/engine/flink/util/transformer/UnionTransformerSpec.scala

```diff
@@ -62,10 +62,35 @@ class UnionTransformerSpec extends FunSuite with BeforeAndAfterAll with Matchers
     MockService.data shouldBe data
   }
 
+  test("should unify streams with union") {
+    val scenario = EspProcess(MetaData("sample-union", StreamMetaData()), NonEmptyList.of[SourceNode](
+      GraphBuilder.source("start-foo", "source")
+        .branchEnd(BranchFooId, UnionNodeId),
+      GraphBuilder.source("start-bar", "noopSource")
+        .branchEnd(BranchBarId, UnionNodeId),
+      GraphBuilder
+        .branch(UnionNodeId, "union", Some(OutVariableName),
+          List(
+            BranchFooId -> List(
+              "value" -> "{a: '#input'}"
+            ),
+            BranchBarId -> List(
+              "value" -> "{a: '123'}"
+            )
+          )
+        )
+        .processorEnd("end", "mockService", "all" -> s"#$OutVariableName.a")
+    ))
+
+    val data = List("10")
```

Comment:
Those union tests are a bit simplistic (testing only one branch?) Maybe we can make `BaseSampleConfigCreator` accept a `Map[SourceId, Data]` and use it to test multiple sources?